src/Sentry.Serilog/SentrySink.cs

```diff
@@ -74,8 +76,18 @@ public void Emit(LogEvent logEvent)
             }
 
             var exception = logEvent.Exception;
-            var formatted = logEvent.RenderMessage(_options.FormatProvider);
             var template = logEvent.MessageTemplate.Text;
+            string formatted;
+            if (_options.TextFormatter is null)
+            {
+                formatted = logEvent.RenderMessage(_options.FormatProvider);
+            }
+            else
+            {
+                var stringWriter = new StringWriter(new StringBuilder());
+                _options.TextFormatter.Format(logEvent, stringWriter);
+                formatted = stringWriter.ToString();
+            }
```

Comment:
since `TextFormatter` is mutable, another way to write this could be:
```suggestion
            if (_options.TextFormatter is { } formatter)
            {
                using var stringWriter = new StringWriter(new StringBuilder());
                formatter.Format(logEvent, stringWriter);
                formatted = stringWriter.ToString();
            }
            else
            {
                formatted = logEvent.RenderMessage(_options.FormatProvider);
            }
```